internal/core/items.rs

```diff
@@ -126,6 +126,8 @@ pub struct ItemVTable {
         core::pin::Pin<VRef<ItemVTable>>,
         orientation: Orientation,
         window_adapter: &WindowAdapterRc,
+        self_component: &vtable::VRc<ItemTreeVTable>,
+        self_index: u32,
```

Comment:
why not `self_rc: &ItemRc,` like the other functions in this vtable have?

Reply:
Ah yes, this is quite annoying. The issue is that `&ItemRc` translates to `*const ItemRc` in the C signature. This is the only function that's called from the C++ code straight into the vtable where a `&ItemRc` is needed. But because this takes a pointer, I can't write `...->layout_info(..., &ItemRc{{ ... })` as that's taking the address of a temporary. In all the other places in the FFI we also pass this "pair".

I could alternatively change this to `&ItemRc` but then on the C++ code call a helper function in our private header files that stores the `ItemRc` on the stack and passes the address. Would you prefer that?

Reply:
> I could alternatively change this to &ItemRc but then on the C++ code call a helper function in our private header files that stores the ItemRc on the stack and passes the address. Would you prefer that?

Yes, I think it is cleaner, don't you thik?